src/metabase/models/card.clj

```diff
@@ -93,16 +93,31 @@
   (fn [_card target-schema-version]
     target-schema-version))
 
+(def transform-metadata-analysis-state
+  "Transform the numeric enum giving the state to and from keywords."
+  (let [kw->num {:not-started 1 ; The default value!
+                 :priority    2
+                 :analyzed    3
+                 :executed    4
+                 :failed      5
+                 :blocked     6}
+        num->kw (into {} (map (fn [[k v]] [v k])) kw->num)]
+    {:in  (merge kw->num                        ; Handle keywords
+                 (update-keys kw->num str)      ; or strings
+                 {nil (kw->num :not-started)})  ; or nil
+     :out #(num->kw % :not-started)}))
```

Comment:
I suggest rather mapping to `:unknown` and then explicitly re-analyze this state as well. That way we can log a warning around encountering them, since logging in this function could emit a lot of spam.

Reply:
Since `:not-started` is the default in the database schema and the column is not nullable, I don't think it's even possible for this not to be defined.

Reply:
One case is rollilng back from a newer version that introduced a new value. Even if that's not likely, and re-analyzing is still the behavior we'd want, I prefer the small amount of ceremony to make this behavior explicit and visible.